pkg/operator/controller/controller_dns_daemonset.go

```diff
@@ -61,6 +62,7 @@ func desiredDNSDaemonSet(dns *operatorv1.DNS, clusterIP, clusterDomain, coreDNSI
 	name := DNSDaemonSetName(dns)
 	daemonset.Name = name.Name
 	daemonset.Namespace = name.Namespace
+	daemonset.SetOwnerReferences([]metav1.OwnerReference{dnsOwnerRef(dns)})
```

Comment:
Can we do an owner reference between the cluster-scoped dns and the namespace-scoped daemonset?

Reply:
Yes, cluster-scoped owners are okay: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#ownerreference-v1-meta